src/background_thread.c

```diff
@@ -368,6 +368,7 @@ check_background_thread_creation(tsd_t *tsd,
 
 		pre_reentrancy(tsd, NULL);
 		int err = background_thread_create_signals_masked(&info->thread,
+			/* NOLINTNEXTLINE(performance-no-int-to-ptr) */
 		    NULL, background_thread_entry, (void *)(uintptr_t)i);
```

Comment:
Given how popular this pattern is, maybe add a wrapper like `void *int_to_ptr_nolint()` and put the comment inside?

Reply:
If you want, sure. But I personally feel like the ugliness of this is a feature, not a bug. The lint (and the act of suppressing the lint) _should_ make you feel icky: smuggling random integers in pointers we should try to avoid when possible.

Reply:
Never mind, I think you're right and this can be improved, I will address this.

Reply:
Sorry forgot to follow up here. I agree there should be some intentional ugliness, but perhaps in the form of a extra verbose function name, then within the wrapper function we can add detailed comments about why the hassle. Another factor is, some functions have a long list of args, and a wrapper would apply to the exact argument instead of the line.

Reply:
How do you feel about meeting in the middle (see latest force-push)? I've extracted the vast majority of offending casts into macros (instead of using magic numbers), which dramatically cuts down the number of `NOLINTNEXTLINE` comments we need.